src/ngraph/runtime/cpu/builder/add.cpp

```diff
@@ -79,6 +79,10 @@ namespace ngraph
             }
 
             REGISTER_OP_BUILDER(Add);
+
+#ifdef NGRAPH_CPU_STATIC_LIB_ENABLE
```

Comment:
This seems to be very fragile i.e., the same #ifdef repeated for every op. Instead consider moving this inside the REGISTER_OP_BUILDER macro or add some other macro?

Reply:
@avijit-nervana this is temporary workaround, i will make a follow up PR where we can build the backend with `--Wl,--whole-archive
  $<TARGET_PROPERTY:cpu_backend,INTERFACE_LINK_LIBRARIES>
  -Wl,--no-whole-archive` option. i tried this but have some issue with static initialization of data structure. I will debug and provide a fix for this in the follow up PR's

Reply:
Could just always register, static or dynamic. Future PR experiments.

Reply:
Will fix later on. But yes, I want to remove global variable initializations.